### PR TITLE
Fix NormalizeWhitespace for switch expressions

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes issue reported on discord.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxNormalizerTests.cs [hunk 1]
```diff
@@ -940,7 +940,7 @@ private static void TestNormalizeStatement(string text, string expected)
         {
             var node = SyntaxFactory.ParseStatement(text.NormalizeLineEndings());
             var actual = node.NormalizeWhitespace("  ").ToFullString();
-            Assert.Equal(expected.NormalizeLineEndings(), actual.NormalizeLineEndings());
+            AssertEx.Equal(expected.NormalizeLineEndings(), actual.NormalizeLineEndings());
```
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxNormalizerTests.cs:943]: Produces nicer error messages when strings are wrong (like a diff of how they're wrong).

File: src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxNormalizerTests.cs [hunk 2]
```diff
@@ -5882,11 +5882,38 @@ void M()
                 """);
         }
 
+        [Fact]
+        public void TestNormalizeSwitchExpressionAsArgument()
+        {
+            TestNormalizeStatement(
+                """writer.Write(data switch{global::Core.Networking.Data.NetworkedVector3 => 0, _ => throw new System.Exception("Failed to handle type in autogenerated polymorphic serialization!")});""",
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxNormalizerTests.cs:5889]: Honestly, my preference is to keep the test as is.  It reflects the customer's use case.  So if any part of it regresses, we'll know.  I'm happy to add more cases.  But i've found that more often than not, whittling down cases actually can be a detriment as it's often a confluence of interactions that can lead to subtle bugs.

[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxNormalizerTests.cs:5910]: tests added.

**Comment by CyrusNajmabadi** [comment]
@CatSandwich can you take this PR over?  Thanks!

**Comment by CatSandwich** [comment]
> @CatSandwich can you take this PR over? Thanks!

I'm not sure that I'm acquainted enough with this ecosystem to do so.

**Comment by CyrusNajmabadi** [comment]
Ok.  Closing out.  If you are interested in picking this up in the future, feel free to do so.

**Pull request closed without merging**